libvirt/resource_libvirt_dir_pool.go

```diff
@@ -0,0 +1,240 @@
+package libvirt
+
+import (
+	"encoding/xml"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	libvirt "github.com/libvirt/libvirt-go"
+	"github.com/libvirt/libvirt-go-xml"
+)
+
+func resourceLibvirtDirPool() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceLibvirtDirPoolCreate,
+		Read:   resourceLibvirtDirPoolRead,
+		Delete: resourceLibvirtDirPoolDelete,
+		Exists: resourceLibvirtDirPoolExists,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"path": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"capacity": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"allocation": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"available": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+				ForceNew: true,
+			},
+			"xml": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"xslt": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+					},
+				},
+			},
+		},
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+	}
+}
+
+func resourceLibvirtDirPoolCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*Client)
+	if client.libvirt == nil {
+		return fmt.Errorf(LibVirtConIsNil)
+	}
+
+	poolName := d.Get("name").(string)
+
+	client.poolMutexKV.Lock(poolName)
+	defer client.poolMutexKV.Unlock(poolName)
+
+	// Check whether the storage pool already exists. Its name needs to be
+	// unique.
+	if _, err := client.libvirt.LookupStoragePoolByName(poolName); err == nil {
+		return fmt.Errorf("storage pool '%s' already exists", poolName)
+	}
+	log.Printf("[DEBUG] Pool with name '%s' does not exist yet", poolName)
+
+	poolPath := d.Get("path").(string)
+	poolDef := libvirtxml.StoragePool{
+		// We only support "dir" type at the moment
```

Comment:
is this comment still relevant in our new format? :thinking:

Reply:
not :)